src/App.tsx

```diff
@@ -1,26 +1,71 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './app/store';
+import { getTodos } from './api';
+import { setTodoList } from './features/todos';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const todos = useSelector((state: RootState) => state.todos);
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const { status, query } = useSelector((state: RootState) => state.filter);
+  const dispatch: AppDispatch = useDispatch();
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos()
+      .then(data => {
+        dispatch(setTodoList(data));
+      })
+      .catch(e => {
+        throw new Error(e);
```

Comment:
Throwing an error inside a catch block will not handle the error gracefully. It would be better to log the error or display a user-friendly message instead.